src/hotspot/share/opto/castnode.hpp

```diff
@@ -32,15 +32,21 @@
 //------------------------------ConstraintCastNode-----------------------------
 // cast to a different range
 class ConstraintCastNode: public TypeNode {
+public:
+  enum DependencyType {
+    RegularDependency, // if cast doesn't improve input type, cast can be removed
+    StrongDependency,  // leave cast in even if _type doesn't improve input type, can be replaced by stricter dominating cast if one exist
+    VeryStrongDependency // leave cast in unconditionally
```

Comment:
How about directly using something like `UnconditionalDependency` instead of `VeryStrongDepedendency` to better distinguish it?

Reply:
Done. I agree the naming scheme I chose is not great but I haven't found anything better so far.